### ISSUE TITLE
Fails to run: width 0 requested without left and right anchors

### ISSUE THREAD

**Issue by Nexion** [gergo-salyi/multibg-sway#8]
Probably due to some recent system upgrade, multibg-sway fails to run now:

```
Protocol error 1 on object zwlr_layer_surface_v1@11: width 0 requested without setting left and right anchors
thread 'main' panicked at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/multibg-sway-0.1.7/src/main.rs:163:13:
Failed to read Wayland events: Protocol error 1 on object zwlr_layer_surface_v1@11: width 0 requested without setting left and right anchors
stack backtrace:
   0:     0x559504bb8375 - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::had81234ff7435cf1
   1:     0x559504bdfd1b - core::fmt::write::h926a6eab0b7db89e
   2:     0x559504bb596f - std::io::Write::write_fmt::h0a78402e0af4ef31
   3:     0x559504bb9671 - std::panicking::default_hook::{{closure}}::h6a57805ba14dd967
   4:     0x559504bb934c - std::panicking::default_hook::h7846f8c587a75876
   5:     0x559504bb9cd1 - std::panicking::rust_panic_with_hook::h1cfcd0a9cfc3ba18
   6:     0x559504bb9b37 - std::panicking::begin_panic_handler::{{closure}}::h648cdc2bfac2338a
   7:     0x559504bb8839 - std::sys::backtrace::__rust_end_short_backtrace::h433aaf6cf13c04c2
   8:     0x559504bb97c4 - rust_begin_unwind
   9:     0x5595047c5c33 - core::panicking::panic_fmt::h4b48e04606ec57a9
  10:     0x559504852265 - multibg_sway::handle_wayland_event::hbb7a432fa2bbdf76
  11:     0x55950485180b - multibg_sway::main::h87f72140be461ee3
  12:     0x559504812973 - std::sys::backtrace::__rust_begin_short_backtrace::hf25c9aae33264493
  13:     0x559504854459 - std::rt::lang_start::{{closure}}::hff0f221df6de9fc6
  14:     0x559504bae0c0 - std::rt::lang_start_internal::h4374b4a93e6b20ad
  15:     0x559504853115 - main
  16:     0x7fabde67d3ee - <unknown>
  17:     0x7fabde67d4a9 - __libc_start_main
  18:     0x5595047c6445 - _start
  19:                0x0 - <unknown>
```

**Comment by gergo-salyi** [comment]
I couldn't reproduce this yet. Could you please post your output here with `RUST_LOG=trace` environment varible set, eg. from a shell:
```
RUST_LOG=trace multibg-sway ~/my_wallpapers
```

**Comment by Nexion** [comment]
```
[2024-09-30T18:21:39Z DEBUG sctk] Bound new global [30] wp_viewporter v1
[2024-09-30T18:21:39Z DEBUG sctk] Bound new global [48] wl_output v4
[2024-09-30T18:21:39Z DEBUG sctk] Bound new global [8] zxdg_output_manager_v1 v3
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Argb8888
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Xrgb8888
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Xbgr8888
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Abgr8888
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Bgr888
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Rgbx4444
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Rgba4444
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Rgbx5551
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Rgba5551
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Rgb565
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Xbgr2101010
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Abgr2101010
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Xbgr16161616f
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Abgr16161616f
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Xbgr16161616
[2024-09-30T18:21:39Z DEBUG sctk] supported wl_shm format Abgr16161616
[2024-09-30T18:21:39Z DEBUG multibg_sway::wayland] New output, name: eDP-1, resolution: 1920x1080, integer scale factor: 1, logical size: 1920x1080
[2024-09-30T18:21:39Z DEBUG multibg_sway::wayland] Output 'eDP-1' needs no scaling
[2024-09-30T18:21:39Z DEBUG multibg_sway::wayland] Using pixel format: Bgr888
[2024-09-30T18:21:39Z DEBUG multibg_sway::wayland] Loaded 6 wallpapers on new output for workspaces: _default, 0: term, 9: sysmon, 8: docker, 2: www, 4: mail
[2024-09-30T18:21:39Z DEBUG multibg_sway::wayland] Shm slot pool size for output 'eDP-1' after loading wallpapers: 48600 KiB
[2024-09-30T18:21:39Z DEBUG multibg_sway::wayland] New sum of shm slot pool sizes for all outputs: 48600 KiB
[2024-09-30T18:21:39Z TRACE mio::poll] registering event source with poller: token=Token(0), interests=READABLE
Protocol error 1 on object zwlr_layer_surface_v1@11: width 0 requested without setting left and right anchors
thread 'main' panicked at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/multibg-sway-0.1.7/src/main.rs:163:13:
Failed to read Wayland events: Protocol error 1 on object zwlr_layer_surface_v1@11: width 0 requested without setting left and right anchors
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by gergo-salyi** [comment]
For the record I couldn't reproduce this by testing only on my Arch Linux with its sway 1.9 and wlroots 0.18.1 official packages.

So I tried to fix this based on the error message. Also updated the dependencies.

I released version 0.1.8 with the changes. Test it please and tell me here if it works now

**Comment by Nexion** [comment]
Yes it helped, thanks a lot!

**Comment by gergo-salyi** [comment]
I'm glad. This was my fault then, previously the program didn't do something as a wayland protocol doc says that it should (now fixed by commit b203a152222397aa4f63a0b67d332ca91fa14952).

Thank you for reporting and testing. This bug was probably about to appear for me and others too.
Closing then.
